x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/SessionUtilsTests.java

```diff
@@ -172,6 +178,5 @@ private BlockFactory blockFactory(long maxBytes) {
     private static void enqueueBlock(BytesRefBlock.Builder builder, List<Page> pages) {
         Block block = builder.build();
         pages.add(new Page(block));
-        Releasables.close(builder);
```

Comment:
Change only makes sure `Releasables.close(builder)` is always called for a created builder.
The instance when this didn't happen was when the if-condition within the loop applied just as the while-condition no longer did. In this case a new `builder` was created, but not released.